test/CompressedBlockPrefilteringTest.cpp

```diff
@@ -565,3 +573,656 @@ TEST_F(TestPrefilterExprOnBlockMetadata, testWithOneBlockMetadataValue) {
+      pr(notExpr(orExpr(le(VocabId(1000)), eq(VocabId(1005)))),
+         Variable{"?city"}),
+      pr(andExpr(ge(IntId(10)), neq(IntId(50))), varX),
+      pr(notExpr(le(IntId(10))), varY),
+      pr(andExpr(ge(IntId(10)), le(IntId(100))), varZ));
+  // ?x >= 10 OR (?x >= -10 AND ?x < 0.00)
+  // expected prefilter pairs:
+  // {<((>= 10) OR ((>= -10) AND (< 0.00))), ?x>}
+  evalAndEqualityCheck(
+      orSprqlExpr(geSprql(varX, IntId(10)),
+                  andSprqlExpr(geSprql(varX, IntId(-10)),
+                               ltSprql(varX, DoubleId(0.00)))),
+      pr(orExpr(ge(IntId(10)), andExpr(ge(IntId(-10)), lt(DoubleId(0.00)))),
+         varX));
+  // !(!(?x >= 10) OR !!(?x >= -10 AND ?x < 0.00))
+  // expected prefilter pairs:
+  // {<!(!(>= 10) OR !!((>= -10) AND (< 0.00))), ?x>}
+  evalAndEqualityCheck(
+      notSprqlExpr(orSprqlExpr(
+          notSprqlExpr(geSprql(varX, IntId(10))),
+          notSprqlExpr(notSprqlExpr(andSprqlExpr(
+              geSprql(varX, IntId(-10)), ltSprql(varX, DoubleId(0.00))))))),
+      pr(notExpr(orExpr(
+             notExpr(ge(IntId(10))),
+             notExpr(notExpr(andExpr(ge(IntId(-10)), lt(DoubleId(0.00))))))),
+         varX));
+  // ?y != ?x AND ?x >= 10
+  // expected prefilter pairs:
+  // {<(>= 10), ?x>}
+  evalAndEqualityCheck(
+      andSprqlExpr(neqSprql(varY, varX), geSprql(varX, IntId(10))),
+      pr(ge(IntId(10)), varX));
+  evalAndEqualityCheck(
+      andSprqlExpr(geSprql(varX, IntId(10)), neqSprql(varY, varX)),
+      pr(ge(IntId(10)), varX));
+}
+
+//______________________________________________________________________________
+// For this test we expect that no PrefilterExpression is available.
+TEST(SparqlExpression, getEmptyPrefilterFromSparqlRelational) {
+  const Variable var = Variable{"?x"};
+  const Iri iri = I("<Iri>");
+  const Literal lit = L("\"lit\"");
+  evalAndEqualityCheck(leSprql(var, var));
+  evalAndEqualityCheck(neqSprql(iri, var));
+  evalAndEqualityCheck(eqSprql(var, iri));
+  evalAndEqualityCheck(neqSprql(IntId(10), DoubleId(23.3)));
+  evalAndEqualityCheck(gtSprql(DoubleId(10), lit));
+  evalAndEqualityCheck(ltSprql(VocabId(10), BoolId(10)));
+  evalAndEqualityCheck(geSprql(lit, lit));
+  evalAndEqualityCheck(eqSprql(iri, iri));
+  evalAndEqualityCheck(orSprqlExpr(eqSprql(var, var), gtSprql(var, IntId(0))));
+  evalAndEqualityCheck(orSprqlExpr(eqSprql(var, var), gtSprql(var, var)));
+  evalAndEqualityCheck(andSprqlExpr(eqSprql(var, var), gtSprql(var, var)));
+}
+
+//______________________________________________________________________________
+// For the following more complex SparqlExpression trees, we also expect an
+// empty PrefilterExpression vector.
+TEST(SparqlExpression, getEmptyPrefilterForMoreComplexSparqlExpressions) {
+  const Variable varX = Variable{"?x"};
+  const Variable varY = Variable{"?y"};
+  const Variable varZ = Variable{"?z"};
+  // ?x <= 10.00 OR ?y > 10
```

Comment:
1. 
Can you please create a separate ...Test.cpp file for the tests that test the creation of the prefilterThings from SparqlExpressions.
2. Can you give the test suite a unique name that refers to the filename? It is hard to find tests when there is 4 different files where we have `TEST(SparqlExpression, ...)` in them.


So overall there's (at least) three files:

1. The tests for the SparqlExpressions (already present).
2. The file for the Prefiltering without the expressions (this file before the PR + the clone and format and equal tests)
3. The (new) file, that tests the combination of SparqlExpressions with the prefilterExpressions (test cases already there, so this shouldn't be too much work).